project/data_analysis/python/montecarlo/mc_get_spectra.py

```diff
@@ -126,11 +129,11 @@
                 window_tuple = (win_T, win_pol)
                 del win_T, win_pol
 
-                print("reading window in %.02f s" % (time.time()-t1))
+                print(f"  [split {k}] Reading window in {time.time()-t1:.02f} s")
 
                 t1 = time.time()
                 split = sph_tools.alm2map(signal_alms[ar] + noise_alms[ar], templates[sv])
-                print("alm2map in %.02f s" % (time.time()-t1))
+                print(f"  [split {k}] alm2map in {time.time()-t1:.02f} s")
 
                 t1 = time.time()
                 if (window_tuple[0].pixel == "CAR") & (apply_kspace_filter):
```

Comment:
kspace filter is only relevant for `CAR` pixellisation so I think you should update the `apply_kspace_filter` by doing something like
```
apply_kspace_filter &= window_tuple[0].pixell == "CAR"
``` 
and then use it in the `if` statement 
```
if apply_kspace_filter:
...
```